src/net.cpp

```diff
@@ -835,9 +835,10 @@ int CNetMessage::readHeader(const char *pch, unsigned int nBytes)
         return -1;
     }
 
-    // reject messages larger than MAX_SIZE
-    if (hdr.nMessageSize > MAX_SIZE)
+    // reject if message has an invalid header
+    if (!hdr.IsValid(Params().MessageStart())) {
         return -1;
+    }
```

Comment:
Previously this wouldn't log and now this logs. This is a minor change in behaviour and opens the possibility to fill in the (currently empty) string in `assert_debug_log` in the invalid message functional test.